app/lib/import_export.rb

```diff
@@ -146,40 +147,26 @@ def update_headers(data)
   end
 
   # Finds the symptoms needed for the reports columns
-  def update_assessment_symptom_data(data, patients)
+  def update_assessment_symptom_data(data, jurisdiction)
     # Don't update if assessment symptom data isn't needed
-    return unless data.dig(:assessments, :checked)&.include?(:symptoms)
+    return [] unless data.dig(:assessments, :checked)&.include?(:symptoms)
 
     data[:assessments][:checked].delete(:symptoms)
     data[:assessments][:headers].delete('Symptoms Reported')
 
-    # For large queries (e.g. 100k patients) of the `Symptom.where(...) query below
-    # we saw MySQL give up on queries and throw an exception.
-    # The workaround for this is to collect the symptom ids in a set and find them
-    # using batches of patients.
-    symptoms = Set.new
-    patients.pluck(:id).in_groups_of(10_000, false) do |batch|
-      Symptom.where(
-        condition_id: ReportedCondition.where(
-          assessment_id: Assessment.where(
-            patient_id: batch
-          ).pluck(:id)
-        ).pluck(:id)
-      ).where.not(label: nil).where.not(name: nil).distinct.pluck(:name, :label).each { |symptom| symptoms.add(symptom) }
-    end
-    symptom_names_and_labels = symptoms.to_a.sort { |a, b| a[1] <=> b[1] }.transpose
+    # Query symptom names and labels
+    symptom_names_and_labels = Symptom.where(condition_id: ThresholdCondition.where(threshold_condition_hash: jurisdiction.subtree.pluck(:threshold_hash)))
```

Comment:
The reason why the previous code operated from the patient's assessments backwards is because the previous assessments would have thrershold_conditions with threshold_condition_hashes that might not represent a jjurisdictions _current_ thresholdcondition. Here the code `threshold_condition_hash: jurisdiction.subtree.pluck(:threshold_hash))` would need to return the **current** jurisdiction.subtree threshold_hashes **in addition to** all historic threshold_hashes from within the jurisdiction.subtree.

Reply:
noted, I realize that I will need to think of some other workaround

Reply:
what do you think about this solution: https://github.com/SaraAlert/SaraAlert/pull/930/commits/16a73388c9e00ca9216ff4bae8be17067706a860#diff-c34521fcd0fdc2034d456886aae8fd35299fe908ab88d8e49c60d0b4d39ab26bR158-R159